### PR TITLE
Update GitHub labels

### PR DESCRIPTION
**Pull request by kachkaev**
TODO after approval but before merging: sync [GitHub labels](https://github.com/blockprotocol/blockprotocol/labels) with labels in Yaml

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1202922734858534

### CODE REVIEW

File: .github/workflows/labeler.yml [hunk 1]
```diff
@@ -6,6 +6,9 @@ jobs:
   triage:
     runs-on: ubuntu-20.04
     steps:
-      - uses: actions/labeler@v3
+      - uses: kachkaev/labeler@d89797c51d07680aec17049cc6790e9d323d9a93
```
Comment (kachkaev) [inline .github/workflows/labeler.yml:9]: https://docs.github.com/en/actions/creating-actions/about-custom-actions#using-a-commits-sha-for-release-management

File: .github/labeler.yml [hunk 2]
```diff
@@ -1,104 +1,69 @@
+"area: crates":
+  - "crates/**/*"
```
Comment (nonparibus) [inline .github/labeler.yml:2]: We want to rename `crates` to `libs` in line with our agreed [monorepo structuring principles](https://www.notion.so/hashintel/Monorepo-Structure-216b3150cfdf48bb9f1bc149dd25861b), and additionally move the contents of `packages` in here (except where things may in fact be `apps`).

File: .github/labeler.yml [hunk 3]
```diff
@@ -1,104 +1,69 @@
...
   - "packages/**/*"
-  - "packages/**/.*"
+
+"area: rfcs":
+  - "rfcs/**/*"
 
 "area: site":
-  - "site/*"
-  - "site/.*"
   - "site/**/*"
```
Comment (nonparibus) [inline .github/labeler.yml:26]: We should make `site` a sub-folder within `apps`

File: .github/labeler.yml [hunk 4]
```diff
@@ -1,104 +1,69 @@
...
-  - "**/patches/*"
-  - "**/patches/.*"
   - "**/patches/**/*"
-  - "**/patches/**/.*"
   - "**/yarn.lock"
 
 "area: hub":
-  - "hub/*"
-  - "hub/.*"
   - "hub/**/*"
```
Comment (nonparibus) [inline .github/labeler.yml:9]: Please liaise with @CiaranMn to remove the `hub` folder in its entirety now we support `npm`-based publishing. 

Side-note: designs for a Git-based publication flow are also complete (but the corresponding dev work not yet planned). In any case, we don't want to be maintaining inbound PRs from users containing their blocks if we can avoid it. The `hub` folder and its contents should be removed once their contents have been mirrored into our block-serving solution.

File: .github/labeler.yml (hunk 2 above)
Comment (kachkaev) [reply .github/labeler.yml:2]: The goal of this PR is to sync existing labels with existing repo structure. I suggest syncing labels again ones (if?) `crates` and `packages` are put under `lib`. Might be worth checking this idea again with @CiaranMn and @Alfred-Mountfield in early Oct.

File: .github/labeler.yml (hunk 3 above)
Comment (kachkaev) [reply .github/labeler.yml:26]: That's in scope for https://github.com/blockprotocol/blockprotocol/pull/609 (see TODO in PR description)

File: .github/labeler.yml (hunk 2 above)
Comment (nonparibus) [reply .github/labeler.yml:2]: There's no 'if', as our monorepo structure/principles were agreed. Unless there's new information we didn't have then, let's not re-open that finished discussion. Happy for this change to be made separately, but it does need making.

**Review by nonparibus — APPROVED** [review]
Thanks, Alex

**Pull request merged**